### PR TITLE
added proof of concept for CLI integration tests

### PR DESCRIPTION
**Pull request by nwilbert**

### CODE REVIEW

**Comment by Airblader** [comment]
LGTM, but I'll let @achimh3011 take a look at it since he is a bit more versed in Python than me. :)

File: src/jiradeps/main.py [hunk 1]
```diff
@@ -140,3 +140,7 @@ def log_issues(headline, issues):
     log.info('\n'.join([headline]
                        + ['    {} - {}'.format(issue.key, issue.fields.summary)
                           for issue in issues]))
+
+
+if __name__ == '__main__':
```
Comment (Achimh3011) [inline src/jiradeps/main.py]: Why do you put everything into a `src` directory but then want to directly run from the main module?

File: tests/jiradeps/test_nxdeps.py [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-        fields = MagicMock()
+def test_data(epics, stories):
 
     return {
-        'TEST-123': [test_story()]
+        epics[0].key: stories
     }
 
 
 def test_trivial_graph(test_data):
```
Comment (Achimh3011) [inline tests/jiradeps/test_nxdeps.py]: Would be more clear to remove the `test_data` fixture entirely.

File: src/jiradeps/main.py [hunk 3]
```diff
@@ -32,7 +32,7 @@
               default=os.path.join(os.getcwd(), CONFIG_FILENAME),
               help='Config file to be used '
                    f'(defaults to CWD/{CONFIG_FILENAME})')
-@click.option('--password', help='Jira password')
+@click.option('--password', '-p', help='Jira password')
```
Comment (Achimh3011) [inline src/jiradeps/main.py]: While I approve of the short option, it is quite surprising given the title of the PR.

File: tests/jiradeps/test_main.py [hunk 4]
```diff
@@ -38,14 +46,16 @@ def test_cli(mocker, config, epics, stories):
...
+    result = click_runner.invoke(main.cli,
+                                 ['-f', 'test', '-p', 'test', '-g', 'ABC-123'],
+                                 catch_exceptions=False)
     assert result.exit_code == 0
     assert 'loaded 1 epics' in result.output
     assert 'loaded 2 stories' in result.output
-    assert 'rendered graph as' in result.output
+
+    assert os.path.isfile('test.gv.svg')
+    with open('test.gv.svg') as output_file:
```
Comment (Achimh3011) [inline tests/jiradeps/test_main.py]: Where is this file written? Would it make sense to apply the [`tmpdir` standard fixture](https://docs.pytest.org/en/latest/builtin.html#builtin-fixtures-function-arguments) here?

File: setup.py [hunk 5]
```diff
@@ -36,8 +36,7 @@
         'Environment :: Console',
         'Intended Audience :: Information Technology',
         'License :: OSI Approved :: Apache Software License',
-        'Operating System :: POSIX',
-        'Operating System :: MacOS',
+        'Operating System :: OS Independent',
```
Comment (Achimh3011) [inline setup.py]: Again quite independent from the PR title

Comment (nwilbert) [reply setup.py]: True. However, the integration test in this PR allowed me to test it under Windows.

File: src/jiradeps/main.py (hunk 3 above)
Comment (nwilbert) [reply src/jiradeps/main.py]: Well, it is tested with the integration test ;-)

File: src/jiradeps/main.py (hunk 1 above)
Comment (nwilbert) [reply src/jiradeps/main.py]: Agreed, I'll remove it. At some point it was useful for testing.

File: tests/jiradeps/test_nxdeps.py (hunk 2 above)
Comment (nwilbert) [reply tests/jiradeps/test_nxdeps.py]: True, I'll remove it.

**Pull request merged**